ph-stx-engine/src/main/java/net/sf/joost/stx/Value.java

```diff
@@ -169,10 +167,34 @@ public Value (final Object obj)
               number = ((Number) obj).doubleValue ();
             }
             else
-            {
-              type = OBJECT;
-              object = obj;
-            }
+              if (obj instanceof Collection) {
+                Collection collection = ((Collection) obj);
+                if(collection.isEmpty()) {
```

Comment:
If it's empty, should you still do the iteration further down? If not, then you could add this condition on line 170 so it goes through the else 🤔

Reply:
yup good catch. Updated